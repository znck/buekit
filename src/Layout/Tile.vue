<script>
import { style, styleResolver } from '../utils'
import PropTypes from '@znck/prop-types'

const _isInt = /^[\d]+$/

export default {
  name: 'Tile',
  functional: true,
  props: {
    size: PropTypes.oneOfType(PropTypes.number, PropTypes.string)
      .validate(
        value => (size => 0 < size && size <= 12)(Number(value))
      ),
    ancestor: PropTypes.bool.value(false),
    parent: PropTypes.bool.value(false),
    child: PropTypes.bool.value(false),
    vertical: PropTypes.bool.value(false)
  },
  render (h, ctx = {}) {
    const s = styleResolver(ctx.$style)
    const { size, tag, ancestor, parent, child, vertical } = ctx.props

    return style(
      h(tag, ctx.data, ctx.children),
      s('tile'),
      size !== '' && s('is-' + size),
      ancestor && s('is-ancestor'),
      parent && s('is-parent'),
      child && s('is-child'),
      vertical && s('is-vertical')
    )
  }
}
</script>

<style lang="scss" module>
@import '~bulma/sass/utilities/_all.sass';
@import '../_base.scss';
@import '~bulma/sass/grid/tiles.sass';
</style>

