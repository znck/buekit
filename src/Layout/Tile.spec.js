import { shallow } from 'vue-test-utils'
import { createRenderer } from 'vue-server-renderer'

import Tile from './Tile.vue'

describe('Tile.vue', () => {
  it ('matches snapshot', () => {
    const wrapper = shallow(Tile, {})
    const renderer = createRenderer()

    renderer.renderToString(wrapper.vm, (err, str) => {
      if (err) throw new Error(err)
      expect(str).toMatchSnapshot()
    })
  })
})
