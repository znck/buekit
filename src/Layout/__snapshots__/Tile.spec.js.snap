// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tile.vue matches snapshot 1`] = `<!---->`;
